### PR TITLE
fix #1736: using correctly sorted deps for OCAML_LIBS

### PR DESCRIPTION
**Pull request by gerdstolpmann**
I hope this code is right.

### CODE REVIEW

File: src/plugins/omake/OMakeEquip.ml [hunk 1]
```diff
@@ -331,6 +331,20 @@ let get_lib_deps ?(transitive=false) ?(filter = fun _ -> true) pkg bs =
        bs.bs_build_depends
     )
 
+let get_sorted_trans_lib_deps pkg bs =
```
Comment (gildor478) [inline src/plugins/omake/OMakeEquip.ml]: The code doesn't check for duplicates. I think you may want to get rid of the duplicates.

**Comment by gildor478** [comment]
Can you add a test based on expomake?

File: src/plugins/omake/OMakeEquip.ml (hunk 1 above)
Comment (gerdstolpmann) [inline src/plugins/omake/OMakeEquip.ml]: Can this really happen? This function is mainly using OASISGraph.topological_sort, which is only returning a node when it is first visited in a depth-first traversal of the graph. Unless I'm overlooking something, the list of nodes will only contain at most one instance of a node.

**Comment by gerdstolpmann** [comment]
Unit test added.

File: src/plugins/omake/OMakeEquip.ml (hunk 1 above)
Comment (gildor478) [inline src/plugins/omake/OMakeEquip.ml]: Indeed, I didn't your piece of code right.

**Pull request merged**